charmhelpers/contrib/openstack/vaultlocker.py

```diff
@@ -37,9 +37,28 @@ def __init__(self, secret_backend=None):
         )
 
     def __call__(self):
+        import hvac
+        ctxt = {}
+        # NOTE(hopem): see https://bugs.launchpad.net/charm-helpers/+bug/1849323
         db = unitdata.kv()
+        # get last-token (back-compat)
         last_token = db.get('last-token')
+        # currently known-good secret-id
         secret_id = db.get('secret-id')
+        # log of tokens we see in this iteration
+        current_tokens = ["sentinel"]
```

Comment:
+1. I think the whole logic to control current_tokens can be removed since this variable is supposed to store the values of tokens from each vault unit related, so the condition in line 74 will never be false (because tokens are unique for each vault unit). And at line 126 this list will only be empty in case vault was not initialized and did not provide tokens. After initializing, the list will always have at least one element on each run.

If I'm not missing something, I'm +1 on removing the whole logic and just appending the current tokens to the list.

Reply:
My original rationale for using a sentinel is to differentiate between an empty list that is empty because there are no tokens available and one that is empty because for whatever reason there are missing settings on the  relation that mean that we could not get to read the token i.e. this code:

if all([vault_url, role_id, token]):

Upon review I think it is indeed extra complexity without gain so I will remove it.